### PR TITLE
perf(eomt): reduce instance validation mask memory

### PR DESCRIPTION
**Pull request by KrzysiekSko**
## What has changed and why?

Implements Task 1 from #941 for the DINOv2 and DINOv3 EoMT instance-segmentation validation paths.

Validation resized all query mask logits to the original image size at once. `get_labels_masks_scores` then materializes the boolean masks, their float copy, the sigmoid output and the product, so the peak grows with `num_queries * image_h * image_w`. Training does the same work at the model input size and is an order of magnitude smaller.

The queries are now scored in chunks. Bilinear interpolation and the mask score reduction act on every query alone, so a chunk boundary does not change the result.

The chunk size comes from the training footprint rather than from a fixed byte budget:

```python
chunk_size = max(1, num_queries * resize_pixels // image_pixels)
```

Training scores all queries at the model input size, so a chunk covers the same number of mask elements at the original image size. There is no constant to re-tune when the model input size or the query count changes. Images up to the model input size give a single chunk, which leaves them byte-for-byte unchanged.

Completed masks move to the CPU per chunk, so at most one chunk is resident on the device. Labels and scores stay on their original device for the distributed metric sync.

`get_labels_masks_scores` in `task_model.py` is unchanged.

Refs #941. Supersedes #944.

## How has it been tested?

`tests/_task_models/eomt/test_instance_segmentation.py` covers the chunk arithmetic (training parity, single chunk for small images, the clamp to one), equality with the pre-change pipeline for both the multi-chunk and the single-chunk case, the padded-region crop, the output shapes and the mask device.

End-to-end against the real `get_labels_masks_scores` with 60 queries, a 640x640 model input, a (320, 640) crop and an 800x1600 image (chunk size 19, 4 chunks):

| | peak RSS | labels | masks | scores |
| --- | --- | --- | --- | --- |
| before | 1.73 GiB | — | — | — |
| after | 1.01 GiB | identical | identical | identical |

On CPU all three outputs are bit-identical. `labels` and `masks` are elementwise and stay bit-identical on CUDA too; `scores` sums over the mask pixels, and the CUDA reduction split depends on the outer dimension, so it is mathematically identical but may differ in the last ULP. The tests assert exact equality for labels and masks and `assert_close` for scores.

Not run here: the `torch.cuda.max_memory_allocated()` comparison and the before/after validation run on a real checkpoint, since this machine has no GPU. Both are still worth doing before merge.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [x] Yes
- [ ] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

**Comment by KrzysiekSko** [comment]
Addressed in c21701f and formatted in 6015441.

The chunk budget now uses the larger materialized mask surface by pixel count across `model_image_size` and the final `image_size`, so the first resize cannot bypass the budget when the final image is smaller. I added a regression where `model_image_size=(8, 10)` and `image_size=(4, 5)`; a two-query model-resolution budget produces chunk boundaries `[2, 2, 1]`.

Validation:
- `uv run pytest tests/_task_models/eomt/test_instance_segmentation.py -q`: 3 passed
- focused Ruff check: passed
- focused Ruff format check: passed
- `git diff --check`: passed

**Review by liopeer — APPROVED** [review]
LGTM! Thanks for the contribution and sorry that it took quite some time 🚀
